### PR TITLE
[PLGN-690] - IBM_Qradar -Adding in a connection test and newest sdk version

### PR DESCRIPTION
**Pull request by rbowden-r7**
### Description

Describe the proposed changes:

  - Added in a new connection test 
  - Updated to the newest version of the sdk 
  - Ran refresh
  - Added in extra examples in plugin spec
  - Updated unit tests to validate schemas 

https://rapid7.atlassian.net/browse/PLGN-690

All actions have have been tested via postman with testing being added to the ticket - 

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``rapid7/insightconnect-python-3-38-slim-plugin:{sdk-version-num}`` and ``rapid7/insightconnect-python-3-38-plugin:{sdk-version-num}``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``insight-plugin validate`` which calls ``icon_validate`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `insight-plugin samples`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `insight-plugin validate` and make sure everything passes
2. Run the assessment tool: `insight-plugin run -A`. For single action validation: `insight-plugin run tests/{file}.json -A`
3. Copy (`insight-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/ibm_qradar/plugin.spec.yaml [hunk 1]
```diff
@@ -974,12 +999,19 @@ actions:
         required: false
         description: Specify the list of fields to be returned in the response. Specify the subfields in parentheses. Multiple fields in the same object must be comma separated. Sample fields to filter are id, create_time, username, note_text. More information about the fields can be found in plugin documentation
         example: id, note_text
+      filter:
+        title: Filter
+        type: string
+        required: false
+        description: Restrict the elements in a list based on the contents of various fields. E.g. id=55 and is_deleted = false
```
Comment (cmcnally-r7) [inline plugins/ibm_qradar/plugin.spec.yaml:1006]: You could do away with this if we're already providing an example
```suggestion
        description: Restrict the elements in a list based on the contents of various fields
```

**Review by cmcnally-r7 — CHANGES_REQUESTED** [review]
just this and moving the `request` in each action to a common file like we discussed 😃

File: plugins/ibm_qradar/help.md [hunk 2]
```diff
@@ -619,468 +611,341 @@ Example input:
...
 
 ## Troubleshooting
-
-_This plugin does not contain any troubleshooting information._
+  
+*There is no troubleshooting for this plugin.*
 
 # Version History
 
+* 1.1.0 - Adding in new connection test | Updated the SDK to version 5.4.4 | Ran refresh to upadate all schemas and unit tests
```
Comment (dlaverty-r7) [inline plugins/ibm_qradar/help.md:941]: Typo in update
Can you update to using sdk v5.4.5?

**Review by dlaverty-r7 — APPROVED** [review]

File: plugins/ibm_qradar/help.md (hunk 2 above)
Comment (rbowden-r7) [reply plugins/ibm_qradar/help.md:941]: yeah the version 5.4.5 will be added to this plugin

File: plugins/ibm_qradar/plugin.spec.yaml [hunk 3]
```diff
@@ -8,7 +8,14 @@ support: community
...
 description: IBM QRadar reduces billions of events and flows into a manageable number of actionable offenses that are prioritized by their impact on your business operations. This plugin allows you to use IBM QRadar to orchestrate and automate Ariel search queries and automate offense management
-version: 1.0.0
+key_features: [Start Ariel search, Get Ariel search by ID, Get offenses, Get offense closing reasons, Get assets, Get offense notes, Get offense note by ID, Add note to offense, Update offense, Watch for new offenses]
+links: ['[IBM QRadar](https://www.ibm.com/docs/en/qsip)']
+references: ['[IBM QRadar API Documentation](https://www.ibm.com/docs/en/qradar-common?topic=170-whats-new-in-rest-api-version)', '[AQL](https://www.ibm.com/docs/en/qradar-on-cloud?topic=structure-sample-aql-queries)']
+version: 1.1.0
+connection_version: 1
+version_history: 
+  - '1.1.0 - Adding in new connection test | Updated the SDK to version 5.4.4 | Ran refresh to upadate all schemas and unit tests'
+  - '1.0.0 - Initial plugin'
```
Comment (cmcnally-r7) [inline plugins/ibm_qradar/plugin.spec.yaml:18]: Love this thanks for trying it out 😄

**Review by cmcnally-r7 — APPROVED** [review]

**Review by dlaverty-r7 — APPROVED** [review]

**Comment by dlaverty-r7** [comment]
@rbowden-r7 lgtm, as discussed off line, logic around adding/removing trailing slash as necessary would be great.

**Review by ablakley-r7 — APPROVED** [review]

**Review by cmcnally-r7 — APPROVED** [review]

**Review by llaszuk-r7 — APPROVED** [review]

**Review by igorski-r7 — APPROVED** [review]

**Pull request merged**